src/main/kotlin/org/malachite/estella/queues/AdminQueue.kt

```diff
@@ -0,0 +1,36 @@
+package org.malachite.estella.queues
+
+import com.beust.klaxon.Klaxon
+import org.springframework.amqp.rabbit.connection.CachingConnectionFactory
+import org.springframework.amqp.rabbit.connection.ConnectionFactory
+import org.springframework.amqp.rabbit.core.RabbitAdmin
+import java.io.File
+import java.net.URI
+
+object AdminQueue {
+
+    private val env = System.getenv()
+
+    private val cloudamqpKey = "CLOUDAMQP_URL"
+    const val configPath = "config.json"
+
+    private fun getCloudAmqpFromConfig(): MutableMap<String, String> {
+        val file = File(configPath)
+        if (!file.exists()) throw NoSuchFileException(file)
+        return Klaxon().parse<MutableMap<String, String>>(file)!!
```

Comment:
Mamy trzy różne serializatory. Zdecydujmy się w koncu na któregoś i weźmy go tu używajmy.

Plus czy całe to jest potrzebne? My już ogólnie ładowania konfigu nie mamy w runnerze aplikacji? Jesli tak, to enva można tam załadować, a do funkcji amqpUrl przekazywać przez adnotację @Value (nie napisze Ci teraz, jak to się robi wybacz, bo ogólnie jestem na telefonie): https://www.baeldung.com/spring-value-annotation - punkt 3. Ogólnie tej adnotacji możesz też użyć jako parametr w funkcji, więc nie ma problemu, pamiętaj tylko, że $ to interpolacja w Kotlinie, więc znak $ będziesz musiał wyescapeowac (backslashem \).

Reply:
Yyy jeśli chodzi o serializatory to pamiętam że coś się waliło jak zmienialiśmy z jednego na drugiego XDD